metrics/metrics.go

```diff
@@ -130,12 +130,20 @@ func NewShadowsocksMetrics(ipCountryDB *geoip2.Reader) ShadowsocksMetrics {
 				Help:      "Entries removed from the UDP NAT table",
 			}),
 	}
+}
+
+func NewShadowsocksMetrics(ipCountryDB *geoip2.Reader) ShadowsocksMetrics {
+	m := newShadowsocksMetrics(ipCountryDB)
 	// TODO: Is it possible to pass where to register the collectors?
-	prometheus.MustRegister(m.accessKeys, m.ports, m.tcpOpenConnections, m.tcpProbes, m.tcpClosedConnections, m.tcpConnectionDurationMs,
-		m.dataBytes, m.timeToCipherMs, m.udpAddedNatEntries, m.udpRemovedNatEntries)
+	m.register(prometheus.DefaultRegisterer)
```

Comment:
Let's rename this method to NewPrometheusShadowsocksMetrics and pass the registerer. This way we make it clearer we are using prometheus metrics and stop depending on a global.

Reply:
Done